src/lib/update-lock.ts

```diff
@@ -81,29 +81,28 @@ async function dispose(
 }
 
 /**
- * Composition step used by Supervisor compose module.
- * Take all locks for an appId | appUuid, creating directories if they don't exist.
+ * Take process lock for an appId, then take all service locks of an appId.
+ * Override existing service locks if lockOverride is true.
+ * Return a list of services that were locked and a function that releases the process lock.
+ * Meant to only be used internally as it does not release the process lock if
+ * the lock succeeds.
  */
-export async function takeLock(
+async function takeLockFor(
 	appId: number,
 	services: string[],
 	force: boolean = false,
-) {
-	logger.logSystemEvent(logTypes.takeLock, {
-		appId,
-		services,
-		force,
-	});
-
+): Promise<{ actuallyLocked: string[]; release: () => void }> {
 	const release = await takeGlobalLockRW(appId);
 
-	let lockOverride: boolean;
-	try {
-		lockOverride = await config.get('lockOverride');
-	} catch (err: any) {
-		throw new InternalInconsistencyError(
-			`Error getting lockOverride config value: ${err?.message ?? err}`,
-		);
+	let lockOverride = force;
+	if (!lockOverride) {
+		try {
+			lockOverride = await config.get('lockOverride');
```

Comment:
I know this was there before. But when does `config.get` throw? I feel like we don't take this precaution elsewhere

Reply:
We can leave this as is for now